tests/drivers/uart/uart_elementary/src/main.c

```diff
@@ -0,0 +1,329 @@
+						0x16, 0x17, 0x18, 0x19, 0x20 };
+uint8_t test_buffer[TEST_BUFFER_LEN] = {
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+uint8_t drain_rx_fifo = 0;
+
+#if defined(CONFIG_SETUP_MISMATCH_TEST)
+volatile uint8_t uart_error_counter = 0;
+#endif
+
+#if defined(CONFIG_DUAL_UART_TEST)
+static const struct device *const uart_dev_aux = DEVICE_DT_GET(UART_NODE_AUX);
+uint8_t test_buffer_aux[TEST_BUFFER_LEN] = { 0x00, 0x00, 0x00, 0x00, 0x00,
+					     0x00, 0x00, 0x00, 0x00, 0x00 };
+volatile uint8_t aux_uart_error = 0;
+#endif
+
+/*
+ISR for UART TX action
+*/
+static void uart_tx_interrupt_service(const struct device *dev, int *tx_byte_offset)
+{
+	uint8_t bytes_sent = 0;
+	uint8_t *tx_data_pointer = (uint8_t *)(test_pattern + *tx_byte_offset);
+
+	if (*tx_byte_offset < TEST_BUFFER_LEN) {
+		bytes_sent = uart_fifo_fill(dev, tx_data_pointer, 1);
+		*tx_byte_offset += bytes_sent;
+	} else {
+		*tx_byte_offset = 0;
+		uart_irq_tx_disable(dev);
+	}
+}
+
+/*
+ISR for UART RX action
+*/
+static void uart_rx_interrupt_service(const struct device *dev, uint8_t *receive_buffer_pointer,
+				      int *rx_byte_offset)
+{
+	int rx_data_length = 0;
+	int bytes_received = 0;
+	do {
+		rx_data_length = uart_fifo_read(dev, receive_buffer_pointer + *rx_byte_offset,
+						TEST_BUFFER_LEN);
+		if (drain_rx_fifo == 0) {
+			bytes_received += rx_data_length;
+		}
+	} while (rx_data_length);
+	*rx_byte_offset += bytes_received;
+}
+
+/*
+Callback function for MAIN UART interrupt based transmission test
+*/
+static void interrupt_driven_uart_callback_main_uart(const struct device *dev, void *user_data)
+{
+	int err;
+	static int tx_byte_offset = 0;
+	static int rx_byte_offset = 0;
+	uart_irq_update(dev);
+	err = uart_err_check(dev);
+#if !defined(CONFIG_SETUP_MISMATCH_TEST)
+	zassert_equal(err, 0, "Unexpected UART device: %s error: %d", dev->name, err);
+#else
+	if (err != 0) {
+		uart_error_counter++;
+	}
+#endif
+	while (uart_irq_is_pending(dev)) {
+		if (uart_irq_rx_ready(dev)) {
+			uart_rx_interrupt_service(dev, (uint8_t *)user_data, &rx_byte_offset);
+		}
+		if (uart_irq_tx_ready(dev)) {
+			uart_tx_interrupt_service(dev, &tx_byte_offset);
+		}
+	}
+}
+
+#if defined(CONFIG_DUAL_UART_TEST)
+/*
+Callback function for AUX UART interrupt based transmission test
+*/
+static void interrupt_driven_uart_callback_aux_uart(const struct device *dev, void *user_data)
+{
+	int err;
+	static int tx_byte_offset_aux = 0;
+	static int rx_byte_offset_aux = 0;
+	uart_irq_update(dev);
+	err = uart_err_check(dev);
+#if !defined(CONFIG_COVERAGE)
+	zassert_equal(err, 0, "Unexpected UART device: %s error: %d", dev->name, err);
+#endif
+	while (uart_irq_is_pending(dev)) {
+		if (uart_irq_rx_ready(dev)) {
+			uart_rx_interrupt_service(dev, (uint8_t *)user_data, &rx_byte_offset_aux);
+		}
+		if (uart_irq_tx_ready(dev)) {
+			uart_tx_interrupt_service(dev, &tx_byte_offset_aux);
+		}
+	}
+}
+#endif
```

Comment:
please add in a comment a reference to starting #if statement `/* CONFIG_DUAL_UART_TEST */`